service/storage_service.cc

```diff
@@ -7168,5 +7170,52 @@ future<> storage_service::node_ops_abort_thread() {
     __builtin_unreachable();
 }
 
+static thread_local std::unordered_map<session_id, std::unique_ptr<session>> sessions;
```

Comment:
Since it has its own header may be we should put the implementation into a separate file as well. This one is already huge.